android/app/src/main/java/com/tourneynizer/tourneynizer/adapters/TabFragmentPagerAdapter.java

```diff
@@ -0,0 +1,89 @@
+package com.tourneynizer.tourneynizer.adapters;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.app.FragmentStatePagerAdapter;
+import android.text.Spannable;
+import android.text.SpannableString;
+import android.text.style.ImageSpan;
+import android.util.Log;
+import android.view.ViewGroup;
+
+import com.tourneynizer.tourneynizer.R;
+import com.tourneynizer.tourneynizer.data.User;
+import com.tourneynizer.tourneynizer.fragments.CreateTournamentFragment;
+import com.tourneynizer.tourneynizer.fragments.RootFragment;
+import com.tourneynizer.tourneynizer.fragments.SearchFragment;
+import com.tourneynizer.tourneynizer.fragments.TournamentListFragment;
+import com.tourneynizer.tourneynizer.fragments.UserProfileFragment;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by ryanwiener on 2/9/18.
+ */
+
+public class TabFragmentPagerAdapter extends FragmentPagerAdapter {
+
+    private static final int NUM_TABS = 5;
+
+    private FragmentManager fragmentManager;
+    private RootFragment currentFragment;
+
+    public TabFragmentPagerAdapter(FragmentManager fm) {
+        super(fm);
+        fragmentManager = fm;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        RootFragment rootFragment = RootFragment.newInstance();
+        switch (position) {
```

Comment:
Explain to me the high level purpose of this switch statement in the adapter. Is it possible to avoid switching based off tab position, like have the classes stored in an array somewhere and use that for creating a new fragment instance?

Reply:
Does creating a new instance for each tab mean context is lost when you switch between the two?

Reply:
I'm pretty sure the adapter takes care of this for me and only calls this function if it can't find a fragment for the tab position. When I have tested it, tabs don't lose context.

Reply:
Good. Was just asking. If the array thing is hard, then we can just keep this for now and refactor later.